pkg/porter/editor.go

```diff
@@ -0,0 +1,86 @@
+package porter
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"get.porter.sh/porter/pkg/context"
+)
+
+const (
+	defaultEditor        = "vi"
+	defaultEditorWindows = "notepad"
+)
+
+// Editor displays content to a user using an external text editor, like vi or notepad.
+// The content is captured and returned.
+//
+// The `EDITOR` environment variable is checked to find an editor.
```

Comment:
As a follow-up issue, we can allow the user to set `editor` in our [config file](https://porter.sh/configuration/) and use that before checking the environment variable. I can make an issue to do that once your PR is merged.

Reply:
Sounds good.